### PR TITLE
Forth word index in latex documentation

### PR DESCRIPTION
**Pull request by lonetech**
Builds (somewhat clumsily) HTML, ePub and PDF versions of the documentation, now with an index. 
Intended as a starting point for #448 

Multiple passes are sadly required to get ToC and index right.

### CODE REVIEW

File: docs/manual.tex [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
 \usepackage{hyperref}
 \usepackage{color}
 \usepackage{graphicx}
 \usepackage{eso-pic}
 \usepackage{multicol}
 \usepackage{alltt}
 
+\makeindex[intoc, columns=3]
+
 \hypersetup{pdfauthor="Ravelli/Durex", pdftitle="durexForth: Operators Manual", colorlinks=true}
```
Comment (jkotlinski) [inline docs/manual.tex:13]: ("ravelli" and "durex" are joke names, at some point it should be changed for something appropriate)

File: docs/manual.tex [hunk 2]
```diff
@@ -1,16 +1,34 @@
 \documentclass[a4paper]{report}
 \usepackage{microtype}
+\usepackage{imakeidx}
```
Comment (jkotlinski) [inline docs/manual.tex:3]: How comes `imakeidx` instead of regular `makeidx`?

**Review by jkotlinski — COMMENTED** [review]
I'm not sure I can provide much meaningful feedback. Is it fine to merge?

File: docs/manual.tex (hunk 2 above)
Comment (lonetech) [reply docs/manual.tex:3]: The main known difference is that imakeidx supports multiple indices (e.g. concepts and words). The reason I used it is that imakeidx supports options, in this case to be included in the table of contents and render in three columns. There exist more variants, like index, indextools, robustindex, etc. Basically, I picked the easy path. Alternate rendering tools like hevea and plastex had limited support for other packages than makeidx and imakeidx, but that's probably a non-issue in tex4ht.

File: docs/manual.tex (hunk 1 above)
Comment (lonetech) [reply docs/manual.tex:13]: The quotation marks here also make it into the PDF metadata, where they probably shouldn't be.

**Comment by lonetech** [comment]
I think it should be fine to merge. If we find a better way to do things like the epub index, we'll put that in then. For now, there's more to do and I'd love to have this as a basis.

**Comment by jkotlinski** [comment]
Cool! Thank you for the contribution!

**Comment by jkotlinski** [comment]
@lonetech after getting the changes, I get this error when building the manual. What am I missing?

`Writing index file durexforth.idx
(/usr/share/texmf-dist/tex/latex/l3backend/l3backend-pdftex.def)
(./durexforth.aux) (/usr/share/texmf-dist/tex/latex/microtype/mt-cmr.cfg)
(/usr/share/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texmf-dist/tex/latex/refcount/refcount.sty)
(/usr/share/texmf-dist/tex/generic/gettitlestring/gettitlestring.sty))
(./durexforth.out) (./durexforth.out)
(/usr/share/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/usr/share/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))
! Undefined control sequence.
l.18 \ifdef
           {\Css}{
?
`

**Comment by lonetech** [comment]
Oh my. That's not what we want; the whole point of `\ifdef` is to avoid running undefined control sequences. This could be a TeX distribution difference; I'm currently using texlive 2022.20220722-1 from Debian. [texfaq.org](https://texfaq.org/FAQ-isdef) suggests `\ifdefined`. 

Could you try if https://github.com/jkotlinski/durexforth/commit/573a3597ec8a335aa102ccfa74df2750dbabc221 helps?

**Comment by jkotlinski** [comment]
`\ifdefined` did not help. `\import{etoolbox}` did the trick.

**Comment by jkotlinski** [comment]
@lonetech another problem now. Any idea what might be wrong?

(this uses the script at share/texmf-dist/scripts/make4ht/make4ht)

```
make4ht -e manual.mk4 -j durexforth manual.tex
[STATUS]  make4ht: Conversion started
[STATUS]  make4ht: Input file: manual.tex
This is pdfTeX, Version 3.141592653-2.6-1.40.24 (TeX Live 2022/Built by MSYS2 project) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
...64/share/texmf-dist/scripts/make4ht/make4ht-indexing.lua:123: attempt to index a nil value (local 'f')
make: *** [Makefile:25: durexforth.html] Error 1
```

**Comment by jkotlinski** [comment]
Probably I should simply install full texlive distribution. I will try that.

**Pull request merged**